test/integration/shortcode-converter.test.js

```diff
@@ -64,6 +118,41 @@ describe( 'segmentHTMLToShortcodeBlock', () => {
 <p>Bar</p>` );
 	} );
 
+	it( 'should convert a shortcode to a block type with a passing `isMatch`', () => {
+		const original = `<p>[my-broccoli id="42"]</p>`;
+
+		const transformed = segmentHTMLToShortcodeBlock( original, 0 );
+		const expectedBlock = createBlock( 'test/broccoli', { id: 42 } );
+		expectedBlock.clientId = transformed[ 1 ].clientId;
+		expect( transformed[ 1 ] ).toEqual( expectedBlock );
+	} );
+
+	it( 'should not convert a shortcode to a block type with a failing `isMatch`', () => {
+		const original = `<p>[my-broccoli id="1000"]</p>`;
+
+		const transformed = segmentHTMLToShortcodeBlock( original, 0 );
+		const expectedBlock = createBlock( 'core/shortcode' );
+		expectedBlock.clientId = transformed[ 1 ].clientId;
+		expect( transformed[ 1 ] ).toEqual( expectedBlock );
+	} );
+
+	it( 'should not blindly exclude a transform in subsequent shortcodes after a failed `isMatch`', () => {
+		const original = `<p>[my-broccoli id="1001"]</p>
+		<p>[my-broccoli id="42"]</p>
+		<p>[my-broccoli id="1000"]</p>`;
+
+		const transformed = segmentHTMLToShortcodeBlock( original );
```

Comment:
Could we add some integration tests as well?

Reply:
@ellatrix, a few of questions:

1. How would they differ from the current ones? It's not like any core blocks are using `isMatch`, nor that there are multiple core blocks interested in the same shortcode tag.
2. Looking at [blocks-raw-handling.test.js](https://github.com/WordPress/gutenberg/blob/master/test/integration/blocks-raw-handling.test.js), I notice that there are now three "areas" for testing:
  - individual tests under _Blocks raw handling_
  - the fixture-based tests under _pasteHandler_
  - individual tests under _rawHandler_

Seems like we could revisit this? I personally don't really know where to start adding tests for this.

3. That `readFile` helper silently bails if the provided file path doesn't exist, meaning that `types` in _pasteHandler_ can just silently pass even when there are missing fixtures. Case in point: `caption-shortcode` no longer runs. Seems like something to fix too, I don't see a reason not to throw an error if a file is missing.

Reply:
> It's not like any core blocks are using isMatch, nor that there are multiple core blocks interested in the same shortcode tag.

Interestingly, this could be changed. An example off the top of my head is that the `[video]` shortcode in Core explicitly supports YouTube and Vimeo URLs in the `src` attribute. These shortcodes could be transformed into their respective `core-embed` block, rather than the `core/video` block.

Reply:
> 1. How would they differ from the current ones? It's not like any core blocks are using `isMatch`, nor that there are multiple core blocks interested in the same shortcode tag.

That's true, but we could add a test block?

> Seems like we could revisit this? I personally don't really know where to start adding tests for this.

Yeah, we should clean that up :) Ideally, if applicable, tests should be run for both the raw and paste handler.

That `readFile` helper silently bails if the provided file path doesn't exist, meaning that `types` in _pasteHandler_ can just silently pass even when there are missing fixtures. Case in point: `caption-shortcode` no longer runs. Seems like something to fix too, I don't see a reason not to throw an error if a file is missing.

Me neither. Sounds good to throw an error.